### ISSUE TITLE
Object not shown as needed in new debugger

### ISSUE THREAD

**Issue by eidan66** [facebook/react-native#43838]
### Description

I'm using the new debugger, and when I try to see a string obj, it shows wrong.
Also, I can't copy~past the object.

https://github.com/facebook/react-native/assets/41434778/ca6f1485-79d8-4059-9db3-3f79a0bab9f6

Here is the same body that i parse in the video, but it shows only one character :-(
<img width="546" alt="Screenshot 2024-04-04 at 9 32 10" src="https://github.com/facebook/react-native/assets/41434778/38c5967f-e71f-46df-bd4f-1b110156b90c">

### Steps to reproduce

Install the latest version of RN, run `npx react-native start --experimental-debugger`, `yarn android/ios` and click open the new debugger after the app is load.

### React Native Version

0.73.5

### Affected Platforms

Runtime - Android, Runtime - iOS

### Output of `npx react-native info`

```text
npx react-native info
info Fetching system and libraries information...
System:
  OS: macOS 14.4.1
  CPU: (10) arm64 Apple M2 Pro
  Memory: 432.20 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.10.0
    path: /usr/local/bin/node
  Yarn: Not Found
  npm:
    version: 10.2.5
    path: /usr/local/bin/npm
  Watchman:
    version: 2024.01.22.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.14.3
    path: /Users/idanlev/.gem/ruby/3.3.0/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.4
      - iOS 17.4
      - macOS 14.4
      - tvOS 17.4
      - visionOS 1.1
      - watchOS 10.4
  Android SDK:
    API Levels:
      - "33"
      - "34"
    Build Tools:
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 33.0.1
      - 33.0.2
      - 34.0.0
    System Images:
      - android-33 | Google APIs ARM 64 v8a
      - android-33 | Google Play ARM 64 v8a
      - android-34 | Google APIs ARM 64 v8a
      - android-UpsideDownCakePrivacySandbox | Google Play ARM 64 v8a
      - android-UpsideDownCakePrivacySandbox | Google Play Intel x86_64 Atom
    Android NDK: Not Found
IDEs:
  Android Studio: 2023.2 AI-232.10300.40.2321.11567975
  Xcode:
    version: 15.3/15E204a
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.10
    path: /usr/bin/javac
  Ruby:
    version: 3.3.0
    path: /Users/idanlev/.rubies/ruby-3.3.0/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.5
    wanted: 0.73.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

info React Native v0.73.6 is now available (your project is running on v0.73.5).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.73.6
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.73.6
info For more info, check out "https://reactnative.dev/docs/upgrading?os=macos".
```
```


### Stacktrace or Logs

```text
There is no logs to this problem
```

### Reproducer

https://github.com/eidan66/React-native-debugger-bug

### Screenshots and Videos

<img width="546" alt="Screenshot 2024-04-04 at 9 32 10" src="https://github.com/facebook/react-native/assets/41434778/380fb228-eeaa-4e40-b73a-cf070af38c59">

https://github.com/facebook/react-native/assets/41434778/7c50dcfe-48f7-4c2c-a609-4e242af71c06

**Comment by benoitgirardmonespaceproprio** [comment]
with 0.74.1
debugger, still show ▶ Object instead of ▶ Object { key: "value", key2: "value2" }

copy~paste is working :)

**Comment by cortinico** [comment]
Closing as this has been fixed in 0.74.1 (and also in 0.73.x).
See https://github.com/facebook/react-native/issues/43258
